src/commands/test_three_party.rs

```diff
@@ -0,0 +1,167 @@
+//! `test` - Scenario: Three parties, a manufacturer of prescription medicines,
+//! a purchaser of the medicines, and a supplier of the medicines. The manufacturer issues a ZSA
+//! for every dose of medicine produced. On purchase, the manufacturer transfers the corresponding
+//! number of ZSAs to the purchaser. The purchaser then transfers the ZSAs to the supplier, in
+//! exchange for the physical doses. The supplier burns the ZSAs after receiving them to signal the
+//! sale of the medicines.
+
+use abscissa_core::{Command, Runnable};
+use orchard::keys::Scope::External;
+
+use crate::commands::test_balances::{
+    check_balances, print_balances, update_balances_after_burn, update_balances_after_transfer,
+    BurnInfo, TestBalances, TransferInfo,
+};
+use crate::components::rpc_client::reqwest::ReqwestRpcClient;
+use crate::components::transactions::sync_from_height;
+use crate::components::transactions::{create_issue_transaction, mine};
+use crate::components::user::User;
+use crate::prelude::*;
+
+/// Run the test scenario
+#[derive(clap::Parser, Command, Debug)]
+pub struct TestThreePartyCmd {}
+
+impl Runnable for TestThreePartyCmd {
+    /// Run the `test` subcommand.
+    fn run(&self) {
+        let config = APP.config();
+        let mut rpc_client = ReqwestRpcClient::new(config.network.node_url());
+        let mut wallet = User::random(&config.wallet.miner_seed_phrase);
+
+        wallet.reset();
+
+        let num_users = 3;
+
+        let manufacturer_index = 0;
+        let purchaser_index: u32 = 1;
+        let supplier_index: u32 = 2;
+
+        let manufacturer = wallet.address_for_account(manufacturer_index, External);
```

Comment:
if you are sufixing the indexes, this one should also be suffixed with `_adr` or `_address` or `_ad`. Keep it consistent across the entire script.

Reply:
Done